ldm/dream/devices.py

```diff
@@ -10,15 +10,19 @@ def choose_torch_device() -> str:
         return 'mps'
     return 'cpu'
 
-def choose_autocast_device(device):
-    '''Returns an autocast compatible device from a torch device'''
-    device_type = device.type # this returns 'mps' on M1
-    # autocast only for cuda, but GTX 16xx have issues with it
-    if device_type == 'cuda':
-        device_name = torch.cuda.get_device_name()
+def choose_precision(device) -> str:
+    '''Returns an appropriate precision for the given torch device'''
```

Comment:
@mh-dm;

How about?

```python
    precision = 'float32'
    if device.type == 'cuda':
        device_name = torch.cuda.get_device_name(device)
        if not 'GeForce GTX 1660' in device_name or 'GeForce GTX 1650' in device_name:
            precision = 'float16'
    return precision
```

I'm kind of partial (no pun) to functional programming, and I really like having a single exit point (unless it actually makes the code *more* convoluted)

Reply:
A note that `not` binds stronger than `or` so needs parens: `if not (A or B):`. But I did change towards what you suggested by removing the middle return.
I think I'm impartial to "single exit point" but I do like less code (fewer tokens).

Reply:
Nice, definitely reduces the [McCabe complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity)